kernel/src/process.rs

```diff
@@ -375,6 +379,9 @@ pub enum State {
     /// processes yet. It can also happen if an process is terminated and all
     /// of its state is reset as if it has not been executed yet.
     Unstarted,
+
+    /// The process has a panic and aborted.
```

Comment:
Given that we are discussing using this syscall for cases other than panics in libtock-rs, I suggest something more like "The process has signaled it is permanently finished."